packages/flutter/test/services/text_input_test.dart

```diff
@@ -955,23 +987,52 @@ void main() {
       );
       expectedMethodCalls.add('setStyle');
       expect(control.methodCalls, expectedMethodCalls);
-      expect(fakeTextChannel.outgoingCalls.length, 8);
+      expect(fakeTextChannel.outgoingCalls.length, 9);
       expect(fakeTextChannel.outgoingCalls.last.method, 'TextInput.setStyle');
 
       connection.close();
       expectedMethodCalls.add('detach');
       expect(control.methodCalls, expectedMethodCalls);
-      expect(fakeTextChannel.outgoingCalls.length, 9);
+      expect(fakeTextChannel.outgoingCalls.length, 10);
       expect(fakeTextChannel.outgoingCalls.last.method, 'TextInput.clearClient');
 
       expectedMethodCalls.add('hide');
       final TestWidgetsFlutterBinding binding = TestWidgetsFlutterBinding.ensureInitialized();
       await binding.runAsync(() async {});
       await expectLater(control.methodCalls, expectedMethodCalls);
-      expect(fakeTextChannel.outgoingCalls.length, 10);
+      expect(fakeTextChannel.outgoingCalls.length, 11);
       expect(fakeTextChannel.outgoingCalls.last.method, 'TextInput.hide');
     });
 
+    test('the platform input control receives isMultiline true on attach', () async {
+      final FakeTextInputControl control = FakeTextInputControl();
+      TextInput.setInputControl(control);
+
+      const TextInputConfiguration multilineConfig = TextInputConfiguration(inputType: TextInputType.multiline);
+      const TextInputConfiguration noneConfig = TextInputConfiguration(inputType: TextInputType.none);
+
+      // Test for https://github.com/flutter/flutter/issues/125875.
+      // When there's a custom text input control installed, the platform text
+      // input control receives TextInputType.none and isMultiline flag.
+      // isMultiline flag is set to true when the input type is multiline.
+      // isMultiline flag is set to false when the input type is not multiline.
+      final Map<String, dynamic> noneIsMultilineTrueJson = noneConfig.toJson();
+      final Map<String, dynamic> noneInputType = noneIsMultilineTrueJson['inputType'] as Map<String, dynamic>;
+      noneInputType['isMultiline'] = true;
+
+      final FakeTextInputClient client = FakeTextInputClient(TextEditingValue.empty);
+      TextInput.attach(client, multilineConfig);
+
+      final List<String> expectedMethodCalls = <String>['attach'];
+      expect(control.methodCalls, expectedMethodCalls);
+      expect(control.inputType, TextInputType.multiline);
+      fakeTextChannel.validateOutgoingMethodCalls(<MethodCall>[
+        // When there's a custom text input control installed, the platform text
+        // input control receives TextInputType.none with isMultiline flag
+        MethodCall('TextInput.setClient', <dynamic>[1, noneIsMultilineTrueJson]),
+      ]);
+    }, testOn: 'browser');
```

Comment:
nit: consider using `skip: !kIsWeb` instead

Reply:
Yes, I have changed it to `skip: !kIsWeb`.
Thank you for reviewing.